src/mugs.js

```diff
@@ -832,16 +840,22 @@ define([
                         );
                     }
                     mug.addMessage("nodeID", nameWarning);
+
                     if (!util.isValidElementName(mug.p.nodeID)) {
-                        return util.format(gettext(
+                        return_value = util.format(gettext(
                             "{nodeID} is not a valid Question ID. " +
                             "It must start with a letter and contain only " +
                             "letters, numbers, and '-' or '_' characters."),
                             {nodeID: mug.p.nodeID});
                     } else if (mug.p.nodeID.toLowerCase() === "meta") {
-                        return gettext("'meta' is not a valid Question ID.");
+                        return_value = gettext("'meta' is not a valid Question ID.");
+                    } else if (mug.__originalNodeID && mug.p.nodeID.toLowerCase() !== mug.__originalNodeID) {
+                        changedQuestionIDWarning.message = gettext(
+                            "Changing a Question ID will create a new Question ID (and a new data column). " +
+                            "It will NOT update the existing or previously submitted data.");
                     }
-                    return "pass";
+                    mug.addMessage("nodeID", changedQuestionIDWarning);
+                    return return_value;
```

Comment:
Why was this `return_value` thing added?

Also, it's fine to return `"pass"` here even if the new `nodeID` check failed. I think that may even be preferable since this new message is at "info" level.

Suggestion: undo the `return_value` change and move the `__originalNodeID` check up mimicking the pattern of `nameWarning` above.

Reply:
> Why was this return_value thing added?

The intention is for only show the "info" message if there are no validation errors.  Therefore, it's necessary to include ```mug.addMessage("nodeID", changedQuestionIDWarning);``` after the if-else's that check for invalid question IDs.